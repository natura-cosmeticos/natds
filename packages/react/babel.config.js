const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
