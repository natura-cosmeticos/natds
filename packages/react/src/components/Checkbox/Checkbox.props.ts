type CheckboxVariant = 'standard' | 'indeterminate'

export interface CheckboxProps {

  /**
   * If `true`, the component is checked
   * @default false
   */
  checked: boolean
  /**
   * If `true`, the Checkbox will be disabled.
   * @default false
   */
  disabled?: boolean

  /**
   * The Checkbox variant
   * @default standard
   */
  variant?: CheckboxVariant

  /**
   * The id of the Checkbox element
   */
  id: string

  /**
   * The onChange event occurs when the element is checked.
   *
   */
  onChange: React.ChangeEventHandler<HTMLInputElement>,
}
