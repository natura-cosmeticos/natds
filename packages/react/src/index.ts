export * from './ThemeProvider'
export { useTheme, createUseStyles } from 'react-jss'

// Components
export * from './components/Avatar'
export * from './components/Badge'
export * from './components/Button'
export * from './components/Card'
export * from './components/Checkbox'
export * from './components/Counter'
export * from './components/Dialog'
export * from './components/Divider'
export * from './components/Icon'
export * from './components/IconButton'
export * from './components/Input'
export * from './components/Label'
export * from './components/Link'
export * from './components/ListItem'
export * from './components/Logo'
export * from './components/ProgressIndicator'
export * from './components/RadioButton'
export * from './components/Rating'
export * from './components/TextField'
