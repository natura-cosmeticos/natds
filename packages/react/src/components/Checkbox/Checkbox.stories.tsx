import React, { useState } from 'react'
import { Story, Meta } from '@storybook/react'
import { Checkbox, CheckboxProps } from '.'
import StoryContainer from '../../helpers/StoryContainer'

const componentStatus = `
---

**NOTE FOR UXs**: This component is available with the following attribute status:

  - ✅ **Disabled**
  - ✅ **Indeterminate**

---
`

export default {
  title: 'Components/Checkbox',
  component: Checkbox,
  parameters: {
    componentSubtitle: 'Checkboxes allow users to select one or more items from a set.',
    docs: { description: { component: componentStatus } },
    actions: { argTypesRegex: '^on.*' }
  }
} as Meta

export const Playground: Story<CheckboxProps> = ({ checked, ...args }) => {
  const [check, setCheck] = useState(checked)

  return (
    <Checkbox {...args} checked={check} onChange={() => setCheck(!check)} />
  )
}
Playground.args = {
  checked: false,
  label: 'Example'
}

export const Variants: Story<CheckboxProps> = (args) => {
  const [check1, setCheck1] = useState(true)
  const [check2, setCheck2] = useState(true)

  return (
    <StoryContainer>
      <Checkbox {...args} checked={check1} onChange={() => setCheck1(!check1)} />
      <Checkbox {...args} checked={check2} onChange={() => setCheck2(!check2)} indeterminate />
    </StoryContainer>
  )
}
Variants.args = { ...Playground.args }

export const Disabled: Story<CheckboxProps> = (args) => (
  <StoryContainer>
    <Checkbox {...args} />
    <Checkbox {...args} checked />
    <Checkbox {...args} checked indeterminate />
  </StoryContainer>
)
Disabled.args = {
  ...Playground.args,
  disabled: true
}
