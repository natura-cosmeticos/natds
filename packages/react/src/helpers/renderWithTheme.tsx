import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
