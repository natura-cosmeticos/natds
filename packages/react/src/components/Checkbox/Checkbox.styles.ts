import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { CheckboxProps } from './Checkbox.props'

const styles = createUseStyles((theme: Theme) => ({
  wrapper: {
    height: theme.size.semiX,
    width: theme.size.semiX,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  checkbox: {
    appearance: 'none',
    border: `2px solid ${theme.color.mediumEmphasis}`,
    borderRadius: 2,
    cursor: 'pointer',
    display: 'inline-block',
    height: theme.size.standard,
    margin: 0,
    position: 'relative',
    width: theme.size.standard,
    '&:focus': {
      outline: 'none'
    },
    '&:disabled': {
      border: `2px solid ${theme.color.lowEmphasis}`,
      cursor: 'default'
    },
    '&:checked': {
      backgroundColor: theme.color.primary,
      border: `2px solid ${theme.color.primary}`,
      color: theme.color.surface,
      '&:after': {
        content: ({ indeterminate }: CheckboxProps) => (indeterminate ? '"\uea5d"' : '"\uea3c"'),
        display: 'block',
        fontFamily: 'natds-icons',
        fontSize: theme.size.standard,
        left: '50%',
        position: 'absolute',
        textAlign: 'center',
        top: '50%',
        transform: 'translate(-50%, -50%)'
      },
      '&:disabled': {
        backgroundColor: theme.color.lowEmphasis,
        border: `2px solid ${theme.color.lowEmphasis}`
      }
    }
  }
}))

export default styles
