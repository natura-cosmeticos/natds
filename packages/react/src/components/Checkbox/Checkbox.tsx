import React from 'react'
import { CheckboxProps } from './Checkbox.props'
import Ripple from '../Ripple'
import styles from './Checkbox.styles'

const Checkbox = ({
  checked = false,
  disabled = false,
  id,
  indeterminate = false,
  onChange,
  testID = `ds-checkbox-${id}`,
  value
}: CheckboxProps): JSX.Element => {
  const { checkbox, wrapper } = styles({ indeterminate })

  const getRippleColor = checked ? 'primary' : 'highlight'

  return (
    <Ripple
      isCentered
      hideOverflow={false}
      disabled={disabled}
      color={getRippleColor}
      animationDuration={500}
      showHover
      focus
    >
      <div className={wrapper}>
        <input
          data-testid={testID}
          checked={checked}
          className={checkbox}
          disabled={disabled}
          id={id}
          onChange={onChange}
          type="checkbox"
          value={value}
        />
        <label htmlFor={id} />
      </div>
    </Ripple>
  )
}

export default Checkbox
