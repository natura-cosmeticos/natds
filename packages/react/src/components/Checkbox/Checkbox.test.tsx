import React from 'react'
import { fireEvent } from '@testing-library/react'
import renderWithTheme from '../../helpers/renderWithTheme'
import Checkbox from './Checkbox'
import { CheckboxProps } from './Checkbox.props'

const defaultProps: CheckboxProps = {
  onChange: () => '',
  id: 'any',
  value: ''
}

describe('Checkbox component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Checkbox {...defaultProps} />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly with check', () => {
    const { styles, component } = renderWithTheme(<Checkbox {...defaultProps} checked />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly as indeterminate', () => {
    const { styles, component } = renderWithTheme(<Checkbox {...defaultProps} indeterminate />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should call onChange', () => {
    const onChangeMock = jest.fn()
    const { component } = renderWithTheme(<Checkbox {...defaultProps} onChange={onChangeMock} />)

    fireEvent.click(component.getByTestId('ds-checkbox-any'))

    expect(onChangeMock).toHaveBeenCalledTimes(1)
  })
})
