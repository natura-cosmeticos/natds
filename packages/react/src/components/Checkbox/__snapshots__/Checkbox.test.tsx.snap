// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render correctly when is checked 1`] = `
Array [
  ".sharedRippleEffect-0-2-5 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6 {
  outline: none;
  position: relative;
  align-self: start;
}
.wrapper-0-2-6:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.ripple-0-2-7 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-8 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-9 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-10 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-11 {
    cursor: pointer;
    display: inline-block;
  }
  .wrapper-d0-0-2-11:focus:after {
    background-color: #f4ab34;
  }
  .wrapper-d0-0-2-11:hover:after {
    background-color: #f4ab34;
    opacity: 0.16;
  }
  .ripple-d1-0-2-12 {
    background-color: #f4ab34;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-13 {  }
  .rippleActive-d3-0-2-14 {
    animation: keyframes-ripple-0-2-10 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #777777;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #bbbbbb;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
    border: 2px solid #f4ab34;
    background-color: #f4ab34;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #bbbbbb;
    background-color: #bbbbbb;
  }
  .checkbox-d0-0-2-4 {  }
  .checkbox-d0-0-2-4:checked {  }
  .checkbox-d0-0-2-4:checked:after {
    content: \\"\\";
  }",
  <div>
    <div
      class="container-0-2-1"
    >
      <div
        class="wrapper-0-2-6 wrapper-d0-0-2-11"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-8 rippleContainer-d2-0-2-13"
        >
          <div
            class="ripple-0-2-7 ripple-d1-0-2-12 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            checked=""
            class="checkbox-0-2-3 checkbox-d0-0-2-4"
            data-testid="ds-checkbox-any"
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Checkbox component should render correctly when is disabled 1`] = `
Array [
  ".sharedRippleEffect-0-2-5 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6 {
  outline: none;
  position: relative;
  align-self: start;
}
.wrapper-0-2-6:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.ripple-0-2-7 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-8 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-9 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-10 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-11 {
    cursor: default;
    display: inline-block;
  }
  .wrapper-d0-0-2-11:focus:after {  }
  .wrapper-d0-0-2-11:hover:after {
    background-color: #000000;
    opacity: 0;
  }
  .ripple-d1-0-2-12 {
    background-color: #000000;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-13 {  }
  .rippleActive-d3-0-2-14 {
    animation: keyframes-ripple-0-2-10 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #777777;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #bbbbbb;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
    border: 2px solid #f4ab34;
    background-color: #f4ab34;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #bbbbbb;
    background-color: #bbbbbb;
  }
  .checkbox-d0-0-2-4 {  }
  .checkbox-d0-0-2-4:checked {  }
  .checkbox-d0-0-2-4:checked:after {
    content: \\"\\";
  }",
  <div>
    <div
      class="container-0-2-1"
    >
      <div
        class="wrapper-0-2-6 wrapper-d0-0-2-11"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-8 rippleContainer-d2-0-2-13"
        >
          <div
            class="ripple-0-2-7 ripple-d1-0-2-12 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            class="checkbox-0-2-3 checkbox-d0-0-2-4"
            data-testid="ds-checkbox-any"
            disabled=""
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Checkbox component should render correctly when is indeterminate 1`] = `
Array [
  ".sharedRippleEffect-0-2-5 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6 {
  outline: none;
  position: relative;
  align-self: start;
}
.wrapper-0-2-6:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.ripple-0-2-7 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-8 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-9 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-10 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-11 {
    cursor: pointer;
    display: inline-block;
  }
  .wrapper-d0-0-2-11:focus:after {
    background-color: #000000;
  }
  .wrapper-d0-0-2-11:hover:after {
    background-color: #000000;
    opacity: 0.16;
  }
  .ripple-d1-0-2-12 {
    background-color: #000000;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-13 {  }
  .rippleActive-d3-0-2-14 {
    animation: keyframes-ripple-0-2-10 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #777777;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #bbbbbb;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
    border: 2px solid #f4ab34;
    background-color: #f4ab34;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #bbbbbb;
    background-color: #bbbbbb;
  }
  .checkbox-d0-0-2-4 {  }
  .checkbox-d0-0-2-4:checked {  }
  .checkbox-d0-0-2-4:checked:after {
    content: \\"\\";
  }",
  <div>
    <div
      class="container-0-2-1"
    >
      <div
        class="wrapper-0-2-6 wrapper-d0-0-2-11"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-8 rippleContainer-d2-0-2-13"
        >
          <div
            class="ripple-0-2-7 ripple-d1-0-2-12 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            class="checkbox-0-2-3 checkbox-d0-0-2-4"
            data-testid="ds-checkbox-any"
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Checkbox component should render correctly with default props 1`] = `
Array [
  ".sharedRippleEffect-0-2-5 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6 {
  outline: none;
  position: relative;
  align-self: start;
}
.wrapper-0-2-6:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.ripple-0-2-7 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-8 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-9 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-10 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-11 {
    cursor: pointer;
    display: inline-block;
  }
  .wrapper-d0-0-2-11:focus:after {
    background-color: #000000;
  }
  .wrapper-d0-0-2-11:hover:after {
    background-color: #000000;
    opacity: 0.16;
  }
  .ripple-d1-0-2-12 {
    background-color: #000000;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-13 {  }
  .rippleActive-d3-0-2-14 {
    animation: keyframes-ripple-0-2-10 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #777777;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #bbbbbb;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
    border: 2px solid #f4ab34;
    background-color: #f4ab34;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #bbbbbb;
    background-color: #bbbbbb;
  }
  .checkbox-d0-0-2-4 {  }
  .checkbox-d0-0-2-4:checked {  }
  .checkbox-d0-0-2-4:checked:after {
    content: \\"\\";
  }",
  <div>
    <div
      class="container-0-2-1"
    >
      <div
        class="wrapper-0-2-6 wrapper-d0-0-2-11"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-8 rippleContainer-d2-0-2-13"
        >
          <div
            class="ripple-0-2-7 ripple-d1-0-2-12 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            class="checkbox-0-2-3 checkbox-d0-0-2-4"
            data-testid="ds-checkbox-any"
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Checkbox component should render correctly with label 1`] = `
Array [
  ".sharedRippleEffect-0-2-5 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6 {
  outline: none;
  position: relative;
  align-self: start;
}
.wrapper-0-2-6:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-6:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.ripple-0-2-7 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-8 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-9 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-10 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-11 {
    cursor: pointer;
    display: inline-block;
  }
  .wrapper-d0-0-2-11:focus:after {
    background-color: #000000;
  }
  .wrapper-d0-0-2-11:hover:after {
    background-color: #000000;
    opacity: 0.16;
  }
  .ripple-d1-0-2-12 {
    background-color: #000000;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-13 {  }
  .rippleActive-d3-0-2-14 {
    animation: keyframes-ripple-0-2-10 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #777777;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #bbbbbb;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
    border: 2px solid #f4ab34;
    background-color: #f4ab34;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #bbbbbb;
    background-color: #bbbbbb;
  }
  .checkbox-d0-0-2-4 {  }
  .checkbox-d0-0-2-4:checked {  }
  .checkbox-d0-0-2-4:checked:after {
    content: \\"\\";
  }
  .label-0-2-15 {
    font-family: Roboto, sans-serif;
    letter-spacing: 0.22px;
  }
  .label-d0-0-2-16 {
    font-size: 14px;
    color: #333333;
  }",
  <div>
    <div
      class="container-0-2-1"
    >
      <div
        class="wrapper-0-2-6 wrapper-d0-0-2-11"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-8 rippleContainer-d2-0-2-13"
        >
          <div
            class="ripple-0-2-7 ripple-d1-0-2-12 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            class="checkbox-0-2-3 checkbox-d0-0-2-4"
            data-testid="ds-checkbox-any"
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
      <label
        class="label-0-2-15 label-d0-0-2-16"
      >
        something
      </label>
    </div>
  </div>,
]
`;
